src/features/surveys/components/SurveyEditor/blocks/BlockWrapper.tsx

```diff
@@ -0,0 +1,38 @@
+import { Box, Card, IconButton } from '@mui/material';
+import { Delete, RemoveRedEye } from '@mui/icons-material';
+import { FC, ReactNode } from 'react';
+
+interface BlockWrapperProps {
+  children: ReactNode;
+  hidden: boolean;
+  onDelete: () => void;
+  onToggleHidden: (hidden: boolean) => void;
+}
+
+const BlockWrapper: FC<BlockWrapperProps> = ({
+  children,
+  hidden,
+  onDelete,
+  onToggleHidden,
+}) => {
+  return (
+    <Box
+      marginBottom={1}
+      sx={{ opacity: hidden ? 0.5 : 1, transition: 'opacity 0.2s' }}
+    >
+      <Card>
+        <Box m={2}>{children}</Box>
+        <Box display="flex" justifyContent="end" m={2}>
+          <IconButton onClick={() => onToggleHidden(!hidden)}>
+            <RemoveRedEye />
+          </IconButton>
+          <IconButton onClick={() => onDelete()}>
+            <Delete />
+          </IconButton>
```

Comment:
Give these icons tooltips with indications of what they do? The trash can is pretty obvious to me at least, but the eye icon a little bit more mysterious. I'm not sure how this is handled in other parts though, if we have tooltips in similar places in other parts of the app?

Reply:
That's a good idea. I don't know if I'm just being lazy (or focused elsewhere) but I feel like merging this now and we can look at that kind of stuff towards the end of the cycle. 😊